src/middlewared/middlewared/plugins/vm/devices/cdrom.py

```diff
@@ -40,9 +43,10 @@ def xml(self, *args, **kwargs):
 
     def _validate(self, device, verrors, old=None, vm_instance=None, update=True):
         path = device['attributes']['path']
-        if not os.path.exists(path):
-            verrors.add('attributes.path', f'Unable to locate CDROM device at {path!r}')
-        elif not self.middleware.call_sync('vm.device.disk_uniqueness_integrity_check', device, vm_instance):
+        check_path_resides_within_volume_sync(
+            verrors, 'attributes.path', path, [p['name'] for p in self.middleware.call_sync('pool.query')]
```

Comment:
Since we only care about imported pools you can avoid expensive call by using `query_imported_fast_impl` from `middlewared.utils.zfs`.